Frontend Development

chrishughandrew opened the pull request:
closes #3
closes #4

## Summary
It was a rushed one, but we've got something that resembles a webapp/website. Although it's not complete/polished, I wanted to through up a PR that could be reviewed in my absence. 

I've already forewarned the project team, but I don't believe that there's loads of time left to make major stylistic changes to the site for this stage of the delivery. (I obviously want the site to look as good as possible, but not at the cost of not getting the API integrated,the site published, and tested in the agreed time - what I've learnt, yet again, is that [_good_ frontend design is hard](https://internetingishard.netlify.app/) and takes a long time to learn and practice). 

Any design/stylistic issues and suggestions that come up in review are good to be noted, _will be appreciated_, and can be taken forward to any/the next phase of delivery.

The site comprises: 
- A range of static pages (based on the Home controller) that are publicly accessible: 
  - Home (`Index.cshtml`)
  - About (essentially FAQs)
  - Privacy (content TBC)
  - Contact (content TBC)
- The Prediction controller pages, which require authenication:
  - Index view (the form to add the patient profile/predictors) with appropriate validation. 
  - Summary view (provides the FIND-AF Risk Score, and outlines the patient profile that the score was created against)


## Minor/Miscellaneous Adjustments
There's a list of minor adjustments to address still, that would be good to look at before this one gets merged:
- [x] Add margin on the bottom of prediciton screens (between button and footer) & About screen (maybe just all screens from the `main` element?) (This is notable on smaller (laptop) size screen (the curse of developer 4k monitor).
- [x] Hack the colour on the accordian headers to something in keeping with the red/black colour scheme
- [x] Grey-out the 'Predict' button after first press (jQuery?) 
- [x] Add content to:
  - [x] Privacy
  - [x] Contact
  - [x] Miscellaneous images (from content document)? 
- [x] Adjust homepage cards (remove if not using)
- [x] Adjust margin/border on footer
- [x] About screen:
  - Full stop on end of first bullet point.
  - "Register your interest"
- [x] Form across two columns (repsonsive)
- [x] Summary (responsive)
- [x] ...
- [x] Search for any `TODO` articfacts in the Code, and address them.


## Ramesh' Feedback
In a truly _Agile_ fashion, I'm trying to get a meeting in with Ramesh to give a flythrough demo, so that he's seeing the progress iteratively. I have some points to raise with him about the site content/design, that do well to be noted here: 
- Page Headings/Card contents (thumbnails images? do you have any/want?)
- Styling (though I can't imagine this has much scope to change significantly if we're going to deliver on time)
- Contact us
  - About (do we want to encourage people to "register their interest"
  - Page  (only have an email address to put on this right now: You have something to "fluff" it out?)
- Other legal pages (Privacy & Copyright) - how do you want this to look?

AdamKeeley commented:
Taken a quick look and I like what I'm seeing!

Agree with all of the minor adjustment points, esp. the greyed out/disabled button. 15-30 seconds to generate a prediction isn't _really_ a long time but it's an age if you don't know what feedback to look for to confirm your click was successful/received. It's especially important as **clicking the Predict button multiple times submits multiple requests to the API, all stored in the Azure Table.** Might be worth considering more than simply disabling the button to prevent request duplication but I think that would be a good place to start.

On a 1920x1080 monitor all items in /Prediction are visible without scroll bars when viewed at 80% scale... might be worth doing something with that if it's easy enough. 

I was able to 'break' it:
- signed in as 'uitake'
- signed out as 'uitake'
- unable to sign back in as 'uitake'; instead the app asks for permissions against 'uolake'
- no option to change which user I could sign in as. Selecting Cancel instead of accepting app permissions against 'uolake' resulted in an Unhandled Exception (https://login.microsoftonline.com/error?code=65004)
- was only able to sign in as 'uitake' after signing in as 'uitake' on another service (eg. https://portal.azure.com)
It's probably an edge case as not a very many users will be signed in to multiple O365 accounts. Given likely effort required to resolve it can probably be handled in documentation/user guides (at least for now).

I like that the accordion defaults with the first one open! Looks pretty.

Could the University of Leeds & LIDA footer items link to the respective web pages?

chrishughandrew commented:
## Changes Summary
- Applied spinner partial and javascript, following [this blog](https://ml-software.ch/posts/creating-a-busy-indicator-for-an-asp-net-core-mvc-application)
- add padding around `body` and `footer` elements to reduce 'cramp'
- customised the accordion elements to fit with red 'theme'
- add links to footer images
- cleaned homepage cards (and remvoed addition cards on request of team) 
- made the prediction form two-column responsive for medium 
- added UKCA and Contact page content. 
- make Summary page content repsonsive for mobile screens
- other miscellaneous commits

## Images 
### Form now responsive 
![FormResponsive](https://github.com/LIDA-Data-Analytics-Team/find-af-webapp/assets/72550656/5642ae97-7e83-49e4-b234-e78bb34e5f6a)

### Red accordian 
![AccordianNowRed](https://github.com/LIDA-Data-Analytics-Team/find-af-webapp/assets/72550656/2fa5bd1d-a6e2-407a-bda1-17b5e52ba1c7)

### Loading spinner
![Loading_spinner_gif](https://github.com/LIDA-Data-Analytics-Team/find-af-webapp/assets/72550656/817ad830-6272-4c98-b096-82c81db31184)

AdamKeeley reviewed (approved):
I haven't gone through the code line by line, I trust you to know what you're doing. I have compiled locally and given it a go though and it's looking good!

I know it isn't any faster than before, but the spinning wheel makes it feel like I'm getting a response more quickly. Deffo a worthy addition. 

The red accordion really pulls the room together and the responsive column shifting is also welcome. 

Love it, great work.
